platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_tasks.adoc

```diff
@@ -130,3 +130,58 @@ include::sample[dir="snippets/bestPractices/useCacheabilityAnnotations/groovy",f
 === Tags
 
 `<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:caching,#caching>>`
+
+[[avoid_provider_get_outside_task_action]]
+== Do not call `get()`` on a Provider outside of a Task action
+:keywords: tasks,inputs-and-outputs
+
+When configuring tasks and extensions do not call link:{javadocPath}/org/gradle/api/provider/Provider.html#get()[`get()`] on a provider, use link:{javadocPath}/org/gradle/api/provider/Provider.html#map(org.gradle.api.Transformer)[`map()`], or link:{javadocPath}/https://docs.gradle.org/nightly/javadoc/org/gradle/api/provider/Provider.html#flatMap(org.gradle.api.Transformer)[`flatMap()`] instead.any
+
+=== Explanation
+
+A provider should be evaluated as late as possible (if evaluated). Calling `get()` leads to evaluating it, which can have multiple side effects on your project:
+
+- The value of the provider becomes an input to configuration, causing potential configuration cache misses.
+- It may be too early, so you don’t actually use the final and correct value of the property. This may lead to painful and hard to debug ordering issues
```

Comment:
```suggestion
- The value may be evaluated too early, meaning you might not be using the final or correct value of the property. This may lead to painful and hard to debug ordering issues.
```